### PR TITLE
Ropes Issue #2 - Save Image Link

### LINKED ISSUE #2: Save Image Link
#### As a user I want to save a link to an image.

__Story__:

Now that we have an awesome landing page for our users to experience it is time
to introduce our first feature! In general, we want to deliver value to our
customers as fast as possible. As a result, we have decided that we want our
users to be able to save links to images, and immediately see that we have
saved the URL to the image.

One could argue that simply saving the image, and not showing it, provides some
value. Yes, you could split this story here. However, because saving image URLs
and viewing those images are both relatively small additions that work well
together, it is acceptable to complete them together. Note, on the other hand,
that we do not yet want to implement the image index. Its addition adds too
much complexity, so we’ll save that for a future story.

In general, feature addition, or stories, should be split into the smallest
deliverable chunk that provides value to your users with the caveat that it
might make sense to combine two, sometimes more, tiny and related features into
a single story.

__Acceptance criteria__:
- [x] The landing page links to the image link submission form.
- [x] The link is entered through a form.
- [x] After the form is saved, the link is persisted in the database.
- [x] After the form is submitted, the user is redirected to a page displaying
  the image.

__Dependencies__:
- Deployment Environment

__Note__: Many image sharing sites will permit one to upload images, or
download images when provided a link. For this project we will only work with
the link, i.e., URL, to various images.

### PR DESCRIPTION
**Pull request by zldavis**
Closes #2

### CODE REVIEW

File: db/migrate/20170601162633_create_images.rb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+class CreateImages < ActiveRecord::Migration[5.0]
+  def change
+    create_table :images do |t|
+      t.string :url, null: false
+      t.timestamps null: false
```
Comment (HaominZhang) [inline db/migrate/20170601162633_create_images.rb]: I think the `timestamps` is by default set to `null: false` so we don't have to worry about it. No need to change tho.

File: Gemfile [hunk 2]
```diff
@@ -48,6 +48,7 @@ end
 group :test do
```
Comment (HaominZhang) [inline Gemfile]: Personally I think gem install could be in its own commit, how do you think?

File: app/controllers/images_controller.rb [hunk 3]
```diff
@@ -1,5 +1,20 @@
 class ImagesController < ApplicationController
```
Comment (HaominZhang) [inline app/controllers/images_controller.rb]: Do you think it makes sense that we change the git history to move this commit before the previous one, because I think create the image should be before we can show it. How do you think?

File: app/controllers/images_controller.rb [hunk 4]
```diff
@@ -1,5 +1,20 @@
...
+  def new
+    @image = Image.new
+  end
+
+  def create
+    image = Image.create(image_params)
```
Comment (HaominZhang) [inline app/controllers/images_controller.rb]: If we use `create`, we will bypass the validation we have in the model. Instead we can use `create!`. And another thing is, do we want to render any corresponding alert to user after saving the image, for success and failure case?

File: test/controllers/images_controller_test.rb [hunk 5]
```diff
@@ -1,6 +1,22 @@
...
 
 class ImagesControllerTest < ActionDispatch::IntegrationTest
+  test 'new renders view with initialized Image model' do
+    get new_image_path
+    assert_response :ok
+    assert_template :new
+    assert assigns(:image).present?
+  end
+
+  test 'create persists new Image and redirects to show' do
```
Comment (HaominZhang) [inline test/controllers/images_controller_test.rb]: Can we also have a test case for invalid saving?

File: Gemfile (hunk 2 above)
Comment (zldavis) [inline Gemfile]: personally I feel like it makes sense to include it in the commit where it is first being used, but do not have a strong opinion either way.

Comment (HaominZhang) [inline Gemfile]: The reason why I think that way is if anything went wrong with the commit which installed the gem, we can simply revert that commit and no other part of the code need to be changed.

File: app/controllers/images_controller.rb (hunk 3 above)
Comment (zldavis) [inline app/controllers/images_controller.rb]: Since we redirect to `show` after creating it seemed logical to implement that first. Otherwise we would have the `create` route implemented without an appropriate response, or add extra work that would later be removed when implementing the `show` route. I'm curious what your thoughts are on this?

**Comment by zldavis** [comment]
I believe I have addressed all your comments.
